medusa/post_processor.py

```diff
@@ -844,18 +844,26 @@ def _should_process(current_quality, new_quality, allowed, preferred):
         :param new_quality: The new quality of the episode that is being processed
         :param allowed: Qualities that are allowed
         :param preferred: Qualities that are preferred
-        :return: True if the quality should be processed, False or None otherwise.
+        :return: Tuple with Boolean if the quality should be processed  and Strng with reason if should process or not
         """
         if new_quality in preferred:
             if current_quality in preferred:
-                return new_quality > current_quality
-            return True
+                if new_quality > current_quality:
+                    return True, 'New quality is higher than current Preferred. Accepting quality'
+                else:
+                    return False, 'New quality is same|lower than current Preferred. Ignoring quality'
+            return True, 'New quality is Preferred'
         elif new_quality in allowed:
             if current_quality in preferred:
-                return False
+                return False, 'Current quality is Allowed but we already have a current Preferred. Ignoring quality'
```

Comment:
If your going to work on this. Might be a good moment to use Exceptions instead of tuples. This line could be rewritten to: 
```python
raise PostProcesQualityException('Current quality is Allowed but we already have a current Preferred. Ignoring quality')
```